pkg/operation/botanist/worker.go

```diff
@@ -119,6 +119,15 @@ func (b *Botanist) DeployWorker(ctx context.Context) error {
 			}
 		}
 
+		settings := &extensionsv1alpha1.Settings{}
+		if workerPool.Settings != nil {
+			settings.MachineDrainTimeout = workerPool.Settings.MachineDrainTimeout
+			settings.MachineHealthTimeout = workerPool.Settings.MachineHealthTimeout
+			settings.MachineCreationTimeout = workerPool.Settings.MachineCreationTimeout
+			settings.MaxEvictRetries = workerPool.Settings.MaxEvictRetries
+			settings.NodeConditions = workerPool.Settings.NodeConditions
+		}
```

Comment:
See comment above; if you would reuse `gardencorev1beta1` struct you can simply pass the `workerPool.Settings` in line 151

Reply:
Yes, that's a great suggestion, thanks. done - https://github.com/gardener/gardener/commit/9aacdeb0348a4d06173bb54d71af652807da0846#diff-873d79d4ef156d5ab5a81e3dce9a0a41R142